addons/im_livechat/static/src/messaging/entity/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -0,0 +1,35 @@
+odoo.define('im_livechat.messaging.entity.MessagingNotificationHandler', function (require) {
+'use strict';
+
+const { registerInstancePatchEntity } = require('mail.messaging.entityCore');
+
+registerInstancePatchEntity('MessagingNotificationHandler', 'im_livechat.messaging.entity.MessagingNotificationHandler', {
+
+    //----------------------------------------------------------------------
+    // Private
+    //----------------------------------------------------------------------
+
+    /**
+     * @override
+     * @param {Object} param0
+     * @param {integer} param0.channelId
+     * @param {integer} param0.partner_id
+     */
+    _handleNotificationChannelTypingStatus(data) {
+        const { channelId, partner_id } = data;
+        const channel = this.env.entities.Thread.insert({ id: channelId, model: 'mail.channel' });
+        let partnerId;
+        if (partner_id === this.env.messaging.publicPartner.id) {
+            // Some shenanigans that this is a typing notification
+            // from website public partner.
+            partnerId = channel.directPartner.id;
```

Comment:
This is probably related to a question I already asked, but why can't the direct partner be the public partner directly?

Reply:
As explained before, different names and countries, so they cannot simply share a single entity.

Reply:
Can we get rid of this?
Feels like there is conflict of interest here between using the directPartner, and in some other places handling parter_id = 0, and in some other places creating a fake partner.

I think all of those cases should be handled the same way, we can discuss which is best if needed.

Reply:
This is as close as "handling them all the same way" as I can think of right now: we receive a typing notification from a livechat channel and it relates to public partner. This is definitely not the global public partner, but instead it refers to the visitor, which is exactly what this work-around does with minimal LOCs.

It uses `directPartner` in a similar fashion to DM chats, because livechat and dm chats are super close in similarities: `directPartner` field determines the recipient of the conversation, and since livechat visitor is basically the recipient of a livechat from operator PoV, it makes sense to also use this field. The current naming of this field `directPartner` sucks a bit, what about renaming it to `correspondent`? This would be a good idea IMHO.